File: PadraoState/PadraoState/Entitades/Mario.cs
using PadraoState.Interfaces;

namespace PadraoState.Entitades
{
    class Mario
    {
        protected MarioState Estado;

        public Mario()
        {
            Estado = new MarioPequeno();
        }

        public void PegarCogumelo()
        {
            Estado = Estado.PegarCogumelo();
        }

        public void PegarFlor()
        {
            Estado = Estado.PegarFlor();
        }

        public void PegarPena()
        {
            Estado = Estado.PegarPena();
        }

        public void LevarDano()
        {
            Estado = Estado.LevarDano();
        }
    }
}

File: state/Desafio/Diego Ozorio/src/SuperMarioState/Mario.java
package SuperMarioState;

public class Mario {
    
    protected MarioState estado;
     
    public Mario() 
    {
        estado = new MarioPequeno();
    }
 
    public void pegaCogumelo() 
    {
        estado = estado.pegaCogumelo();
    }
 
    public void pegaFlor() 
    {
        estado = estado.pegaFlor();
    }
 
    public void pegaPena() 
    {
        estado = estado.pegaPena();
    }
 
    public void levaDano() 
    {
        estado = estado.levaDano();
    }
}
